rbc/omnisci_backend/omnisci_array.py

```diff
@@ -31,7 +35,101 @@ def buffer_extra_members(self):
 
 
 class Array(Buffer):
-    pass
+    """
+    In HeavyDB, an Array of type `T` is represented as follows:
+
+    .. code-block:: C
+
+        {
+            T* data,
+            int64_t size,
+            int8_t is_null
+        }
+
+    Array is a contiguous block of memory that behaves like a python list.
```

Comment:
Python list may be a little too high-level object to make this relation. For instance, python lists can be concatenated while contiguous blocks of memories can't, etc. How about this: "Array holds a contiguous block of memory and it implements a subset of the array protocol."